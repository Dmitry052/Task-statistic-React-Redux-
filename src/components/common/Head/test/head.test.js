import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Head } from './../index';

configure({ adapter: new Adapter() });

describe('<Header />', () => {
  const wrapper = shallow(<Head />);
  it('should contain 1 nav element', () => {
    expect(wrapper.find('nav').length).toBe(1);
  });

  it('should contain 1 div element', () => {
    expect(wrapper.find('nav').length).toBe(1);
  });

  it('should contain 2 a(link) elements', () => {
    expect(wrapper.find('a').length).toBe(2);
  });

  it('should contain node logout', () => {
    expect(wrapper.find('#logout').exists()).toBe(true);
  });

  it('Matches the snapshot Head component', () => {
    const tree = shallow(<Head />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
