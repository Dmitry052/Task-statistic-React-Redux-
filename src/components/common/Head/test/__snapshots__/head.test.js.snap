// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Matches the snapshot Head component 1`] = `
<nav
  className="navbar navbar-default navbar-static-top"
>
  <div
    className="container"
  >
    <a />
    <ul
      className="nav navbar-nav navbar-right"
    >
      <a
        id="logout"
      >
        <i
          aria-hidden="true"
          className="fa fa-sign-out fa-lg"
          onClick={[Function]}
        />
         undefined
      </a>
    </ul>
  </div>
</nav>
`;
